packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -43,18 +43,28 @@ export const useArbQueryParams = () => {
   })
 }
 
+const isMax = (amount: string | null | undefined) =>
+  amount?.toLowerCase() === AmountQueryParamEnum.MAX
+
 // Our custom query param type for Amount field - will be parsed and returned as a string,
 // but we need to make sure that only valid numeric-string values are considered, else return '0'
 // Defined here so that components can directly rely on this for clean amount values and not rewrite parsing logic everywhere it gets used
 const AmountQueryParam = {
   // type of amount is always string | undefined coming from the input element onChange event `e.target.value`
-  encode: (amount: string | null | undefined) => amount,
+  encode: (amount: string | null | undefined) => {
+    const amountRegex = new RegExp(/^\d*(\.\d*)?$/)
+    if (isMax(amount)) {
+      return amount
+    }
+    return amountRegex.test(amount ?? '') ? amount : ''
```

Comment:
Hm, last time I tested this it worked fine, but this regex might have messed it up. I can now enter weird stuff like `00--000001,` in the input.

Maybe instead of maintaining a regex we can do:
```
if (isMax(amount)) {
  return amount
}

return isValidNumber(amount) ? amount : ''
```

We can create `isValidNumber` from the check we use in the decode function, where we check for `isNaN` and negative number.

Reply:
This is a problem with existing production build as well.. 
![image](https://user-images.githubusercontent.com/7558499/196935271-34f49271-3596-4801-bf07-52299b87e31d.png)

Reply:
i noticed it's the number input that allows `--` and `++`
it's an existing issue on production.
i was trying to fix it with setting the input to type string with numeric mode on, but then the value would get set to `max` visibly before the max amount gets loaded, so i reverted back to the type number input field

Reply:
adding onto that
it's a problem with vanilla number fields too (＃`Д´)
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number